src/ReactiveUI/Expression/Reflection.cs

```diff
@@ -118,26 +111,21 @@ public static string ExpressionToPropertyNames(Expression expression)
         /// </summary>
         /// <param name="member">The member info to convert.</param>
         /// <returns>A Func that takes in the object/indexes and returns the value.</returns>
-        public static Func<object, object?[]?, object?>? GetValueFetcherForProperty(MemberInfo? member)
+        public static Func<object?, object?[]?, object?>? GetValueFetcherForProperty(MemberInfo? member)
         {
             if (member is null)
             {
                 throw new ArgumentNullException(nameof(member));
             }
 
-            FieldInfo? field = member as FieldInfo;
+            var field = member as FieldInfo;
```

Comment:
if (member is not FieldInfo field)
{
}

Potentially this is a good candidate for a expression switch.

Reply:
CS0165 rule stops me from using :
           ```
 if (member is not FieldInfo field)
            {
                return (obj, _) => field.GetValue(obj) ?? throw new InvalidOperationException();
            }
```